.github/workflows/docs/build-docs

```diff
@@ -28,7 +28,7 @@ def remove_dir(dirpath):
 
 
 def fix_links(filepath):
-    with open(filepath, "r", encoding="utf8") as f:
```

Comment:
I am still not sure why this change is made, have you done this manual or is ruff doing this changes? I have done a PR with only updating black and that is not doing this change.

Reply:
Same answer as before: https://github.com/CQCL/pytket-qir/pull/57#discussion_r1280597356 Ruff does it with its implementation of the pyupgrade linter.

Copying over:

> It's the default option and [hence unneeded](https://beta.ruff.rs/docs/rules/redundant-open-modes/).

Reply:
I am fine with doing this change, maybe my question was not that clear. Had ruff done this automatically or have you done this change manually? And if ruff has done that, with which command run?

Reply:
`ruff check .` shows all the lint violations. Some of those can be automatically corrected by ruff using `ruff check . --fix`. But in this PR, we also have `pre-commit` set up in such a way that `ruff check --fix` will be automatically called on all files that are being added/modified in the commit. The last sentence is True only if `pre-commit install` has been issued once in your local clone once as mentioned in the README in the development steps. I am happy to show a demo if desired.